packages/store/src/scalarMap.js

```diff
@@ -0,0 +1,89 @@
+// Copyright (C) 2019 Agoric, under Apache license 2.0
+
+// @ts-check
+
+import { assert, details as X, q } from '@agoric/assert';
+import { passStyleOf } from '@agoric/marshal';
+import { mustBeComparable } from '../../same-structure';
+
+const assertKey = key => {
+  harden(key); // TODO: Just a transition kludge. Remove when possible.
+  mustBeComparable(key);
+  const passStyle = passStyleOf(key);
+  switch (passStyle) {
+    case 'bigint':
+    case 'boolean':
+    case 'null':
+    case 'number':
+    case 'string':
+    case 'symbol':
+    case 'undefined':
+    case 'remotable': {
+      return;
+    }
+    case 'copyArray':
+    case 'copyRecord':
+    case 'copyError': {
+      assert.fail(X`composite keys not yet allowed: ${key}`);
+    }
+    // case 'promise': is precluded by `mustBeComparable` above
+    default: {
+      assert.fail(X`unexpected passStyle ${passStyle}`);
+    }
+  }
+};
+
+const assertValue = value => {
+  harden(value); // TODO: Just a transition kludge. Remove when possible.
```

Comment:
Maybe mention this in an issue (could be same issue) too?

Reply:
Yes, same